cpp/include/cudf/detail/join.hpp

```diff
@@ -74,9 +75,9 @@ struct hash_join {
   rmm::device_buffer const _composite_bitmask;  ///< Bitmask to denote whether a row is valid
   cudf::null_equality const _nulls_equal;       ///< whether to consider nulls as equal
   cudf::table_view _build;                      ///< input table to build the hash map
-  cudf::structs::detail::flattened_table
-    _flattened_build_table;  ///< flattened data structures for `_build`
-  map_type _hash_table;      ///< hash table built on `_build`
+  std::shared_ptr<cudf::experimental::row::equality::preprocessed_table>
```

Comment:
```suggestion
  std::unique_ptr<cudf::experimental::row::equality::preprocessed_table>
```

Reply:
This is a `shared_ptr` because that's how this object is created https://github.com/rapidsai/cudf/blob/55ed34778694974f19af71f9be82721b8b8af183/cpp/include/cudf/table/experimental/row_operators.cuh#L694